### PR TITLE
update work annotations

### PR DESCRIPTION
**Pull request by haoqing0110**

### CODE REVIEW

File: utils/work/v1/workapplier/workapplier.go [hunk 1]
```diff
@@ -62,11 +62,14 @@ func (w *WorkApplier) Apply(ctx context.Context, work *workapiv1.ManifestWork) (
...
 	}
 
-	if ManifestWorkSpecEqual(work.Spec, existingWork.Spec) {
+	if ManifestWorkEqual(work, existingWork) {
 		return existingWork, nil
 	}
 
 	oldData, err := json.Marshal(&workapiv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: existingWork.Annotations,
```
Comment (qiujian16) [inline utils/work/v1/workapplier/workapplier.go:71]: should we also compare labels and ownerrefs?

Comment (haoqing0110) [reply utils/work/v1/workapplier/workapplier.go:71]: @qiujian16 I don't see the labels and ownerrefs will change during update. Not sure if I miss some cases and need to compare them?

**Pull request merged**